packages/metadata/src/types.ts

```diff
@@ -17,6 +17,7 @@ export type DecoratorType =
  * @typeparam T Type of the metadata value
  * @typeparam D Type of the decorator
  */
+// tslint:disable-next-line:no-unused
```

Comment:
This rule complains that there is no instance (prototype) method or property that would actually access T or D. Because these parameters are not used in such way, the rule things they should be removed.

On line 35, we are inside a static method and providing T and D values to create a specialized MetadataAccessor class. We are providing values to the two generic parameters, but these values won't be used anywhere inside the class.

Let me try to show this difference on a simpler example.

```ts
// This is similar to defining class MetadataAccessor<T, D>
function foo(unused: string) {
  // no-op
}

// This is similar to calling new MetadataAccessor<X, Y>(key)
foo('bar');
```

The function "foo" declares "unused" argument that's not accessed, therefore linter evaluates "unused" as something that can be removed.

Calling `foo('bar')` provides a value for the argument, but this does not make the argument "used", because the provided value is ignored by the function.

Reply:
Thank you for the detailed example, and

> This rule complains that there is no instance (prototype) method or property that would actually access T or D.

Makes sense 👍